cmd/webhook/main.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2020 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/certificates"
+	"knative.dev/pkg/webhook/configmaps"
+	"knative.dev/pkg/webhook/resourcesemantics"
+	"knative.dev/pkg/webhook/resourcesemantics/defaulting"
+	"knative.dev/pkg/webhook/resourcesemantics/validation"
+
+	"knative.dev/eventing-ceph/pkg/apis/sources/v1alpha1"
+)
+
+var types = map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
+	// List the types to validate
+	v1alpha1.SchemeGroupVersion.WithKind("CephSource"): &v1alpha1.CephSource{},
+}
+
+var callbacks = map[schema.GroupVersionKind]validation.Callback{}
+
+const admissionWebhookName = "ceph-source-webhook"
+
+// NewDefaultingAdmissionController sets up mutating webhook.
+func NewDefaultingAdmissionController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return defaulting.NewAdmissionController(ctx,
+
+		// Name of the resource webhook.
+		fmt.Sprintf("defaulting.webhook.%s.knative.dev", system.Namespace()),
```

Comment:
in others, we have hard coded vals:
https://github.com/knative-sandbox/eventing-github/blob/master/cmd/webhook/main.go#L50

Reply:
I mean, look here:

https://github.com/knative-sandbox/eventing-ceph/pull/5/files#diff-bb67564e21779b97af5b71daf54e367aR17